yt_dlp/extractor/cda.py

```diff
@@ -199,17 +218,20 @@ def _api_extract(self, video_id):
 
     def _web_extract(self, video_id, url):
         self._set_cookie('cda.pl', 'cda.player', 'html5')
-        webpage = self._download_webpage(
-            f'{self._BASE_URL}/video/{video_id}/vfilm', video_id)
+        url = f'{self._BASE_URL}/video/{video_id}'
+        webpage = self._download_webpage(url, video_id)
 
         if 'Ten film jest dostępny dla użytkowników premium' in webpage:
             self.raise_login_required('This video is only available for premium users')
 
+        if re.search(f'video/{video_id}/vfilm', webpage):
+            url += '/vfilm'
```

Comment:
Use `_search_regex` with `default=None`

Searching the html for this seems a bit odd though. Does dirkf's suggestion [here](https://github.com/yt-dlp/yt-dlp/pull/6638#issuecomment-1666840370) not work? (after changing `urlh.geturl()` to `urlh.url`)